modules/gapi/test/common/gapi_parsers_tests_common.hpp

```diff
@@ -0,0 +1,384 @@
+            if (image_id < 0.f)
+            {
+                break;    // marks end-of-detections
+            }
+
+            if (confidence < confidence_threshold)
+            {
+                continue; // skip objects with low confidence
+            }
+            adjustBoundingBoxRef(rc);
+
+            const auto clipped_rc = rc & surface;
+            if (filter_out_of_bounds)
+            {
+                if (clipped_rc.area() != rc.area())
+                {
+                    continue;
+                }
+            }
+            out_boxes.emplace_back(clipped_rc);
+        }
+    }
+
+    void parseSSDWLref(const cv::Mat& in_ssd_result,
+                       const cv::Size& in_size,
+                       const float& confidence_threshold,
+                       const int& filter_label,
+                       std::vector<cv::Rect>& out_boxes,
+                       std::vector<int>& out_labels)
+    {
+        out_boxes.clear();
+        out_labels.clear();
+        const auto &in_ssd_dims = in_ssd_result.size;
+        CV_Assert(in_ssd_dims.dims() == 4u);
+
+        const int MAX_PROPOSALS = in_ssd_dims[2];
+        const int OBJECT_SIZE   = in_ssd_dims[3];
+        CV_Assert(OBJECT_SIZE  == 7); // fixed SSD object size
+        cv::Rect surface({0,0}, in_size), rc;
+        float image_id, confidence;
+        int label;
+        const float *data = in_ssd_result.ptr<float>();
+        for (int i = 0; i < MAX_PROPOSALS; i++)
+        {
+            std::tie(rc, image_id, confidence, label)
+                = unitedParseRef(data + i*OBJECT_SIZE, in_size);
+            if (image_id < 0.f)
+            {
+                break;    // marks end-of-detections
+            }
+
+            if (confidence < confidence_threshold &&
+                filter_label != -1 &&
+                label != filter_label)
+            {
+                continue; // filter out object classes if filter is specified
+            }
+
+            out_boxes.emplace_back(rc & surface);
+            out_labels.emplace_back(label);
+        }
+    }
+private:
+    void adjustBoundingBoxRef(cv::Rect& boundingBox)
+    {
+        auto w = boundingBox.width;
+        auto h = boundingBox.height;
+
+        boundingBox.x -= static_cast<int>(0.067 * w);
+        boundingBox.y -= static_cast<int>(0.028 * h);
+
+        boundingBox.width += static_cast<int>(0.15 * w);
+        boundingBox.height += static_cast<int>(0.13 * h);
+
+        if (boundingBox.width < boundingBox.height)
+        {
+            auto dx = (boundingBox.height - boundingBox.width);
+            boundingBox.x -= dx / 2;
+            boundingBox.width += dx;
+        }
+        else
+        {
+            auto dy = (boundingBox.width - boundingBox.height);
+            boundingBox.y -= dy / 2;
+            boundingBox.height += dy;
+        }
+    }
+
+    std::tuple<cv::Rect, float, float, int> unitedParseRef(const float* it,
```

Comment:
it's better to `const float const` as it was above

Reply:
No, i change it. auto generated const float* const. Now const float*.